src/crosswordgenerator.ts

```diff
@@ -16,138 +17,164 @@ const emptyTile: tileCrossWord = {
         let currentScore: number = await getScore(currentCrossword);
         if(currentScore>score){
             score = currentScore;
             crossword = currentCrossword;
+            answers.forEach((answer) => {
+                currentAnswers.push({
+                    answer: answer.answer,
+                    questNumber: answer.questNumber
+                })
+            })
         }
     }
 
+    while(answers.length>0){
+        removeWord(answers[0],crossword,questions)
+        answers.splice(0,1);
+    }
+    
     let output = crossword[0].map((_, colIndex) => crossword.map(row => row[colIndex]));
 
     return output;
 }
 
-async function simpleCrossWord(answers: answer[]): Promise<tileCrossWord[][]>{
+async function simpleCrossWord(answers: answer[], questions: question[]): Promise<tileCrossWord[][]>{
     return new Promise<tileCrossWord[][]>(async (resolve) => {
         let crossword: tileCrossWord[][] = new Array(rows)
                                            .fill(emptyTile)
                                            .map(() =>
                                              new Array(columns).fill(emptyTile)
                                             );
-    
-        //sortAnswers from large to big
-        answers.sort((a,b) => b.answer.length - a.answer.length);
-
-        //place first answer in the upper right position
-        placeWordHorizontal(answers[0],0,0,crossword);
-        answers.splice(0,1);
         
+        //place first answer in the upper right position
+        let indexOfCurrentAnswer = Math.floor(Math.random() * answers.length)
+        placeWordHorizontal(answers[indexOfCurrentAnswer],0,0,crossword);
+        answers.splice(indexOfCurrentAnswer,1);
+        let tries = 0;
         //place remaining words randomly
-        while(answers.length>0) {
+        while(answers.length>0 && tries < 50) {
             //choose random word
-            
             let indexOfCurrentAnswer = Math.floor(Math.random() * answers.length)
             let currentAnswer = answers[indexOfCurrentAnswer];
+            tries++;
             let placedWord = await tryPlaceWord(currentAnswer,crossword);
             if(placedWord){
-                answers.splice(indexOfCurrentAnswer,1)
+                answers.splice(indexOfCurrentAnswer,1);
+                tries = 0;
             }
         }
-
         resolve(crossword);
     })
 }
 
+async function removeWord(word: answer, crossword:tileCrossWord[][], questions: question[]){
+    let questIndex = -1;
+    questions.forEach((quest, i) => {
+        if(quest.answer == word.answer){
+            questIndex = i;
+        }
+    })
+    startpoints.forEach((tile) => {
+        if(Number(tile.answer)>questIndex+1){
+            tile.answer = String(Number(tile.answer)-1);
+        }
+    })
+    questions.splice(questIndex,1);
+}
+
 async function tryPlaceWord(word: answer, crossword:tileCrossWord[][]):Promise<boolean>{
```

Comment:
This method is currently 60 lines long. As we learned in PSE, good methods should not be much longer than 20 lines.
(Exception: Go, where every single statement can span three lines because of error handling)